tests/backtest/test_simulator.py

```diff
@@ -305,6 +310,78 @@ def test_run_strategies(
     assert trade_requests == [mock_trade, mock_trade2]
 
 
+def test_warning_on_run_strategies(
+    asset_pair: AssetPair,
+    timestamp_asset_price: dict[AssetPair, pd.DataFrame],
+    mocker: MockerFixture,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Trades created from running strategies."""
+    caplog.set_level(logging.WARN)
+
+    # Error on Strategy
+    strategy1 = Mock(side_effect=IndexError("Error Message in Strategy"))
+    type(strategy1).asset_pairs = PropertyMock(return_value=[asset_pair])
+    type(strategy1).lookback = PropertyMock(return_value=pd.Timedelta(days=1))
+    type(strategy1).name = "strategy1"
+
+    mock_composer = Mock()
+    mock_composer.get_range.return_value = dict.fromkeys([asset_pair], None)
+    type(mock_composer).start_time = PropertyMock(
+        return_value=dt.datetime.fromtimestamp(int("runrun", 36))
+    )
+    type(mock_composer).end_time = PropertyMock(
+        return_value=dt.datetime.fromtimestamp(int("runrun", 36))
+    )
+    type(mock_composer).freq = PropertyMock(return_value=pd.DateOffset(days=3))
+
+    portfolio = Mock()
+
+    mocker.patch("coin_test.backtest.Simulator._strategies_to_run")
+    Simulator._strategies_to_run.return_value = [strategy1]
+    mocker.patch("coin_test.backtest.Simulator._validate")
+    Simulator._validate.return_value = True
+
+    mock_slippage_calculator = Mock()
+    mock_transaction_calculator = Mock()
+    sim = Simulator(
+        mock_composer,
+        portfolio,
+        [],
+        mock_slippage_calculator,
+        mock_transaction_calculator,
+    )
+
+    time = pd.Timestamp(dt.datetime.fromtimestamp(int("runrun", 36)))
+    schedule = [(strategy1, croniter("@yearly"))]
+    trade_requests = sim.run_strategies(schedule, time, portfolio)
+
+    strategy1.assert_called_with(time, portfolio, mock_composer.get_range.return_value)
+
+    assert caplog.record_tuples == [
+        (
+            "coin_test.backtest.simulator",
+            logging.WARN,
+            "Strategy strategy1 raised an exception.",
+        ),
+        ("coin_test.backtest.simulator", logging.WARN, "Error Message in Strategy"),
+    ]
+
+    assert trade_requests == []
+
+    sim_raise_blocking_err = Simulator(
```

Comment:
Thinking about it more: this is testing two separate behaviors -- let's have 2 separate tests to be clear about this

Reply:
done